parameters.txt
<<<<<<< HEAD
M_tf 0.935
FL 390
h_cr 11887.2
R 287.05
g 9.80665
gamma 1.4
M_cr 0.79
M_max 0.82
rho_0 1.225
T_0 288.15
s_l 2000.0
V_land 58.485532834
cV 0.024
A 14.0
c 17.0
C_f_e 0.0026
V_s 100.0
V_rot 1.1
n_max 2.5
f 0.9745
P_c 10.9158882525
x_spar1 0.2
x_spar2 0.65
Afactor 1.2
e 0.85
MTOW 67834.0
S 110.0
thrust 213.0
SwetSref 5.5
W_wing 7656.92099792
W_fuselage 10302.1124054
W_landinggear 2041.24019077
W_empennage 1275.47187667
W_propulsion 6384.77149726
W_nacelle 666.628085126
W_equipment 9606.65747111
OEW 37933.8025243
MTOW 665224.2961
MTOW1 69176.5532573
OEW1 39064.0656119
W_empty1 38369.8156119
W_crew 694.25
W_fuel 9529.67798556
payload 20375.28
n_pax_act 226
W_eng 5577.98172757
mdot 628.592177217
Di 2.06853888889
Dh 2.06853888889
Dn 2.35058090615
Def 2.35058090615
Dg 1.83763410879
Deg 1.01069875984
ln 3.25
lf 4.20070028764
Betalf 1.6335911721
h_cabin 2.192
h_cargo 1.195
w_cargo 3300.1974757
N_aisle 1
w_seat 90
w_aisle 20
sidewall_clearance 2
seat_corr 1.129
h_floor 1.378
t_floor 110
h_headspace 63
h_aisle 86.3
h_luggage 23.3
h_extra 0
seat_pitch 28
N_seat_row 6
l_cockpit 4
ratio_tail 1.6
N_passengers 240
ext_int_corr 1.088
N_rows 40
w_floor 3.61
d_int_fus 3.68
d_ext_fus 4
r_int_fus 1.84
r_ext_fus 2.0
l_seats 28.448
l_cabin 33.7288
ratio_nosecone 1.5
ratio_tailcone 3.0
l_nosecone 6.0
l_tailcone 12.0
l_fus 44.1288
F 11.0322
area_cargo 2.849
area_cabin 7.39007608799
area_ext_fus 12.5663706144
l_tail 6.4
lambdac_0 0.573082227315
taper 0.291021465676
b 39.2428337407
c_r 4.34239031253
c_t 1.26372879329
t_c 0.0408643428426
dihedral -2.12200503714
MAC 3.08483965102
M_dd 0.82
c_r_f 4.34239031253
c_t_f 1.42399228673
c_avg 3.0037553146
fuel_in_wings 15.3108
empty_tip 0.93
A_cross_section 0.050685452813
fuel_used ['confuel', 'jatropha']
fuel_mix [20, 80]
fuelvolume_needed 10.1993148585
fuelcost 0.134869895554
emissions_per_s 586.057236244
aromatics 14.8
C_L_cr 0.703501377258
K 0.0222906082762
C_D_cr 0.0253319382907
LoverD 27.6601693403
C_D_0 0.0143
Sh 16.1895210693
Sv 23.9817317304
lv 18.0
sweep_h 33
sweep_v 40
Ah 4.4
Av 1.8
lambda_h 0.5
lambda_v 0.5
b_h 9.07753636789
b_v 6.57016872803
cr_h 2.55758101222
ct_h 1.37538429817
cr_v 4.86679165039
ct_v 2.4333958252
MAC_h 2.13948668604
MAC_v 3.785
YMAC_h 1.87555940896
YMAC_v 1.46003749512
Nmw 28.0
Nnw 2
Nst 4
p 0.174959784442
Pmw 21857.369729
Pnw 26608.971844
z 3
lnlg 16
lm 3.5
ymlg1 3.69357753948
psi 1.09955742876
V_cr 233.103734559
eta_airfoil 0.95
Cl_max_cruise_clean 2.363
alpha_0L_low_clean -0.0666715774262
CL_alpha_low_clean 4.67396959314
alpha_stall_low_clean 0.381447952722
Cd_min_cruise_clean 0.006
Cm_des_cruise_clean -0.116
alpha_0L_cruise_clean -0.0666715774262
alpha_des_cruise_clean 0.0237364778271
alpha_stall_cruise_clean 0.174282044879
alpha_trim_cruise_clean 0.053469509097
CL_max_cruise_clean_plane 0.944327272727
CL_max_cruise_clean_wing 1.03876
CL_max_low_clean 1.76819242385
CL_alpha_cruise_clean 6.06963453989
Cd_min_low_clean 0.005
M_crit_cruise_clean 0.663
lambdac_LE 0.574911356893
lambdac_2 0.517666951232
lambdac_4 0.546787527368
dCL_HLD_land 0.431807576149
dCL_HLD_TO 0.331807576149
CL_0 0.31
A_eff 14.75
e 2.71828182846
CD0_tot 0.0153294049444
dCD_excresence 0.03
S_wet 684.207340796
Swf_LE 81.3311204699
Swf_TE 49.2523844543
S_flaps_land 12.0668341913
S_flaps_to 8.6191672795
S_slats 7.48246308323
bs_i 3.0
bs_o 18.160705832
bf_i 1.0
bf_o 7.73356701313
d_f_land 50
d_f_to 20
c_ext_c_land 1.245
c_ext_c_to 1.175
S_a 1.04477281931
ba_i 15.5009193276
ba_o 18.6403460268
Sh_S 0.13
S_h 14.3
wingpos 0.435
x_lemac_pos 19.196028
front_bar_cg_final 0.132852279934
front_cg_final 19.6058559809
aft_bar_cg_final 0.545485399376
aft_cg_final 20.878762989
cgrange_final 1.27290700818
=======
M_tf 0.935
FL 390
h_cr 11887.2
R 287.05
g 9.80665
gamma 1.4
M_cr 0.79
M_max 0.82
rho_0 1.225
T_0 288.15
s_l 2000.0
V_land 58.485532834
cV 0.024
A 14.0
c 17.0
C_f_e 0.0026
V_s 100.0
V_rot 1.1
n_max 2.5
f 0.9745
P_c 10.9158882525
x_spar1 0.2
x_spar2 0.65
Afactor 1.2
e 0.85
MTOW 67834.0
S 110.0
thrust 213.0
SwetSref 5.5
W_wing 7656.92099792
W_fuselage 10302.1124054
W_landinggear 2041.24019077
W_empennage 1275.47187667
W_propulsion 6384.77149726
W_nacelle 666.628085126
W_equipment 9606.65747111
OEW 37933.8025243
MTOW 665224.2961
MTOW1 69176.5532573
OEW1 39064.0656119
W_empty1 38369.8156119
W_crew 694.25
W_fuel 9529.67798556
payload 20375.28
n_pax_act 226
W_eng 2788.99086379
mdot 314.296088609
Di 1.45542030663
Dh 1.45542030663
Dn 1.67392147816
Def 1.67392147816
Dg 1.16964105734
Deg 0.643302581539
ln 3.25
lf 3.14168619213
Betalf 1.22175601151
h_cabin 2.192
h_cargo 1.195
w_cargo 3300.1974757
N_aisle 1
w_seat 90
w_aisle 20
sidewall_clearance 2
seat_corr 1.129
h_floor 1.378
t_floor 110
h_headspace 63
h_aisle 86.3
h_luggage 23.3
h_extra 0
seat_pitch 28
N_seat_row 6
l_cockpit 4
ratio_tail 1.6
N_passengers 240
ext_int_corr 1.088
N_rows 40
w_floor 3.61
d_int_fus 3.68
d_ext_fus 4
r_int_fus 1.84
r_ext_fus 2.0
l_seats 28.448
l_cabin 33.7288
ratio_nosecone 1.5
ratio_tailcone 3.0
l_nosecone 6.0
l_tailcone 12.0
l_fus 44.1288
F 11.0322
area_cargo 2.849
area_cabin 7.39007608799
area_ext_fus 12.5663706144
l_tail 6.4
lambdac_0 0.573082227315
taper 0.291021465676
b 39.2428337407
c_r 4.34239031253
c_t 1.26372879329
t_c 0.0408643428426
dihedral -2.12200503714
MAC 3.08483965102
M_dd 0.82
c_r_f 4.34239031253
c_t_f 1.42399228673
c_avg 3.0037553146
fuel_in_wings 15.3108
empty_tip 0.93
A_cross_section 0.050685452813
fuel_used ['confuel', 'jatropha']
fuel_mix [20, 80]
fuelvolume_needed 61.0222246357
fuelcost 0.806922933278
emissions_per_s 3506.364576
aromatics 14.8
C_L_cr 0.703501377258
K 0.0222906082762
C_D_cr 0.0253319382907
LoverD 27.6601693403
C_D_0 0.0143
Sh 16.1895210693
Sv 23.9817317304
lv 18.0
sweep_h 33
sweep_v 40
Ah 4.4
Av 1.8
lambda_h 0.5
lambda_v 0.5
b_h 9.07753636789
b_v 6.57016872803
cr_h 2.55758101222
ct_h 1.37538429817
cr_v 4.86679165039
ct_v 2.4333958252
MAC_h 2.13948668604
MAC_v 3.785
YMAC_h 1.87555940896
YMAC_v 1.46003749512
Nmw 28.0
Nnw 2
Nst 4
p 0.174959784442
Pmw 21857.369729
Pnw 26608.971844
z 3
lnlg 16
lm 3.5
ymlg1 3.69357753948
psi 1.09955742876
V_cr 233.103734559
eta_airfoil 0.911890652781
Cl_max_cruise_clean 2.363
alpha_0L_low_clean -0.0666715774262
CL_alpha_low_clean 4.50792998744
alpha_stall_low_clean 0.39538202868
Cd_min_cruise_clean 0.006
Cm_des_cruise_clean -0.116
alpha_0L_cruise_clean -0.0666715774262
alpha_des_cruise_clean 0.0237364778271
alpha_stall_cruise_clean 0.180292327976
alpha_trim_cruise_clean 0.0578477468026
CL_max_cruise_clean_plane 0.944327272727
CL_max_cruise_clean_wing 1.03876
CL_max_low_clean 1.46131605277
CL_alpha_cruise_clean 5.86370696182
Cd_min_low_clean 0.005
M_crit_cruise_clean 0.663
lambdac_LE 0.574911356893
lambdac_2 0.517666951232
lambdac_4 0.546787527368
dCL_HLD_land 0.431807576149
dCL_HLD_TO 0.331807576149
CL_0 0.31
A_eff 14.75
e 2.71828182846
CD0_tot 0.0153294049444
dCD_excresence 0.03
S_wet 684.207340796
Swf_LE 75.9301578508
Swf_TE 83.7811809157
S_flaps_land 17.5940479923
S_flaps_to 14.6617066603
S_slats 6.98557452228
bs_i 4.0
bs_o 18.9232128524
bf_i 2.0
bf_o 16.4832242615
d_f_land 40
d_f_to 15
c_ext_c_land 1.21
c_ext_c_to 1.175
cf_c 0.35
cs_c 0.15
S_a 1.04477281931
ba_i 15.5009193276
ba_o 18.6403460268
Sh_S 0.13
S_h 14.3
wingpos 0.435
x_lemac_pos 19.196028
front_bar_cg_final 0.156616246498
front_cg_final 19.6791640072
aft_bar_cg_final 0.563327831914
aft_cg_final 20.9338040324
cgrange_final 1.25464002522
front_bar_cg_final 0.132852279934
front_cg_final 19.6058559809
aft_bar_cg_final 0.545485399376
aft_cg_final 20.878762989
cgrange_final 1.27290700818


parameters.py
from math import *
import numpy as np
import matplotlib.pyplot as plt
#------------------------------------------------CONVERSION--------------------------------------------------

lbf_to_N = 4.4482216
hr_to_s = 3600.
lbm_to_kg = 0.45359237
ft_to_km = 0.3048/1000
ft_to_m = 0.3048
gal_to_L = 3.78541
FL_to_m = ft_to_m*100
rad_to_deg = 180/pi
inch_to_cm=2.54
cm_to_inch=1/2.54
mm_to_m=1./1000.
m_to_mm=1000.
cm_to_m=1/100.
inch_to_m=inch_to_cm*cm_to_m
rad_to_deg=180/pi
deg_to_rad=pi/180

m_to_cm=1/100.

M_tf = 0.935                        #technology factor for supercritical airfoils
FL = 390                            #flight level
h_cr = FL*FL_to_m                   #cruise altitude
R = 287.05
g = 9.80665                         #gravitational acc.
gamma = 1.4
M_cr = 0.79                         #cruise mach number (req)
M_max = 0.82                        #maximum cruise mach number
rho_0 = 1.225                       #sea-level density
T_0 = 288.15                        #sea-level temperature 
s_l = 2000.                         #landing distance
V_land = sqrt(s_l/0.5847)           #landing velocity based on landing distance
cV = 0.024                          #climb gradient requirement from CS25
A = 14.                             #Aspect ratio
c = 17.                             #climb rate as from ref. aircraft
C_f_e = 0.003                       #friction coefficient
V_s = 100.                          #stall speed based on reference aircraft
V_rot = 1.1                         #rotation speed
n_max = 2.5                         #max load factor (CS25)
f = 0.9745                          #fuel fraction during cruise

mft        = 3.2808399        # conversion factor meters to feet
kglbs      = 2.20462262       # conversion factor kilograms to pounds
sqmsqft    = 10.7639104       # conversion factor square meters to square feet
mskts      = 1.94384449       # conversion factor meters per seconds to knots
degrad     = 0.0174532925     # conversion factor degrees to radians
kmnm       = 0.539956803      # conversion factor kilometers to nautical miles
Nlbs       = 0.224808943      # conversion factor newtons to pounds
papsf      = 0.02089          # conversion factor pascal to pound-force per square foot
kgmlbsgall = 0.00834540445    # conversion factor for kilogram per cubic meter to pounds per gallon
papsi      = 0.000145037738   # conversion factor for pascal to pounds-force per square inch


#------------------------------------------------FUNCTIONS-----------------------------------------------------

#---------PARAMETERS----------

def value(symbol):
    file = 'parameters.txt'
    file = np.genfromtxt(file, dtype=str, delimiter=';')
    lst = []
    found = False

    for i in range(len(file)):
        lst.append(file[i].split())

    for i in range(len(lst)):
        if lst[i][0] == symbol:
            found == True
            return eval(lst[i][1])
    if found == False:
        return "parameter has not been found"
    
#------------ISA--------------

lapselst = [-.0065,0,.001,.0028,0,-.0028,-.002,0.]
hlst = [0,11000,20000,32000,47000,51000,71000,84852,100000]
T0lst = [288.15]
p0lst = [101325]
g = 9.80665
R = 287.05

for i in range(1,8):
    T0 = T0lst[i-1] + lapselst[i-1]*(hlst[i]-hlst[i-1])
    T0lst.append(T0)
    
    if lapselst[i-1] != 0:
        p0 = p0lst[i-1] * (T0lst[i]/T0lst[i-1])**(-g/(lapselst[i-1]*R))
    else:
        p0 = p0lst[i-1] * e**(-g/(R*T0lst[i])*(hlst[i]-hlst[i-1]))

    p0lst.append(p0)
    
def ISA(h):
    for i in range(8):
        if hlst[i] <= h <= hlst[i+1]:
            h0 = hlst[i]
            T0 = T0lst[i]
            p0 = p0lst[i]
            a  = lapselst[i]
            
            if a != 0:
                T = T0 + a*(h-h0)
                p = p0 * (T/T0)**(-g/(a*R))
            else:
                T = T0
                p = p0 * e**(-g/(R*T0)*(h - h0))

            rho = p / (R*T)

    return T,p,rho

def ISA_rho(rho):
    T0 = T0lst[i]
    a  = lapselst[i]
    if rho > ISA(11000)[2]:
        h = (rho/1.225**((-g/(a*R)-1)**-1)-1)*T0/a
#--------------speed------------------

def a(h): #speed of sound

    a = sqrt(value("gamma")*value("R")*ISA(h)[0])

    return a

def max_speed(h):

    V = value("M_max")*a(h)

    return V
    
def cruise_speed(h_cr):

    V_cr = value("M_cr")*a(value("h_cr"))

    return V_cr

#---------dynamic pressure------------

def q(V,h): 

    q = 0.5*ISA(h)[2]*V**2

    return q

def cruise_q(h_cr):
    
    q_cr = 0.5*ISA(value("h_cr"))[2]*cruise_speed(value("h_cr"))**2

    return q_cr

#-------------thrust------------------

def cruise_thrust(h,C_D_cr,S):
    
    T_cr = 0.5*ISA(h)[2]*cruise_speed(h)**2*value("C_D_cr")*value("S")

    return T_cr

